src/D05/BoardingPass.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+
+namespace aoc.D05
+{
+	public class BoardingPass
+	{
+		private int _seatID;
+
+		public BoardingPass(string code)
+		{
+			if (code.Length != 10)
+				throw new ArgumentException("code must be 10 digits long!");
+
+			_seatID = GetSeatID(code);
+		}
+
+		public int SeatID { get { return _seatID; } }
+
+		public static int PartI(BoardingPass[] passes)
+		{
+			int max = -1;
+
+			foreach (var b in passes)
+			{
+				if (b.SeatID > max)
+					max = b.SeatID;
+			}
+
+			return max;
+		}
+
+		public static int PartII(BoardingPass[] passes)
+		{
+			Array.Sort(passes, (x1, x2) => x1.SeatID.CompareTo(x2.SeatID));
+
+			int mySeatID = -1;
+
+			for (int i = 0; i < passes.Length; i++)
+			{
+				if (i != passes.Length - 1) //last element should not be checked
+				{
+					if ((passes[i + 1].SeatID - passes[i].SeatID) != 1)
+					{
+						mySeatID = passes[i].SeatID + 1;
+						break;
+					}
+				}
+			}
+
+			return mySeatID;
+		}
+
+	  private int GetSeatID(string s)
+		{
+			var s_row = s.Substring(0, 7);
```

Comment:
@taherkheli why this variable name? seems like a c-standard

Reply:
changed. how the **** do u find out if something is a c standard?

Reply:
In csharp you don't really use the variable name to signal the type. so anything like `string_x` is not how you would do it